frame/staking/src/tests.rs

```diff
@@ -704,6 +704,107 @@ fn nominating_and_rewards_should_work() {
 		});
 }
 
+#[test]
+fn set_payee_also_updates_payee_destination() {
+	ExtBuilder::default().build_and_execute(|| {
+		// When
+		assert_ok!(Staking::set_payee(RuntimeOrigin::signed(11), RewardDestination::Controller));
+
+		// Then
+		assert_eq!(Payee::<Test>::get(11), RewardDestination::Controller);
+		assert_eq!(
+			Payees::<Test>::get(11),
+			PayoutDestination::Split((Perbill::from_percent(100), 11))
+		);
+	});
+}
+
+#[test]
+fn update_payee_works() {
```

Comment:
Any reason to keep multiple test scenarios in a single test? Each `given, when, then` could be its own test.

Reply:
I can indeed refactor if it means it wins your approval ;)